operator/operator.go

```diff
@@ -189,17 +204,28 @@ func NewOperatorFromConfig(c optypes.NodeConfig) (*Operator, error) {
 		0: "quorum0",
 	}
 	economicMetricsCollector := economic.NewCollector(
-		sdkClients.ElChainReader, sdkClients.AvsRegistryChainReader,
-		AVS_NAME, logger, common.HexToAddress(c.OperatorAddress), quorumNames)
+		sdkClients.ElChainReader,
+		sdkClients.AvsRegistryChainReader,
+		AVS_NAME,
+		logger,
+		common.HexToAddress(c.OperatorAddress),
+		quorumNames,
+	)
 	reg.MustRegister(economicMetricsCollector)
 
+	var optionalMetrics metrics.Metrics
+	if c.EnableMetrics {
+		optionalMetrics = metrics.NewEigenMetrics(AVS_NAME, c.EigenMetricsIpPortAddress, reg, logger)
+	} else {
+		optionalMetrics = metrics.NewNoopMetrics()
+	}
+
 	operator := &Operator{
-		config:     c,
-		logger:     logger,
-		ethClient:  ethRpcClient,
-		metricsReg: reg,
-		// TODO: replace with noop in case not enabled
-		metrics:                    sdkClients.Metrics,
+		config:                     c,
+		logger:                     logger,
+		ethClient:                  ethHttpClient,
+		metricsReg:                 reg,
+		metrics:                    optionalMetrics,
```

Comment:
Solves a pending TODO by only constructing relevant metrics when they're enabled in config.